### PR TITLE
feat: add getAllOpList

### PR DESCRIPTION
**Pull request by demohi**

### CODE REVIEW

**Comment by demohi** [comment]
#18

**Comment by ry** [comment]
Rather than manually copying in all these protos (which may become out of date with the so library) is there some other way to link to these? Maybe having TF be a git-submodule? Or maybe we can get TF to distribute the proto files in its binary tarballs, since they are apparently part of the public API.

**Comment by yorkie** [comment]
> Rather than manually copying in all these protos (which may become out of date with the so library) is there some other way to link to these?

What my previous thoughts is downloads those proto files from GitHub raw files, and remove them from source.

> Or maybe we can get TF to distribute the proto files in its binary tarballs, since they are apparently part of the public API

@ry current binary dist doesn't contain the proto files, maybe this a Pull Request for TensorFlow?

File: lib/index.js [hunk 1]
```diff
@@ -110,3 +111,12 @@ tensorflow.session = function() {
...
 }
+
+/**
+ * @method getAllOpList
+ */
+
+ tensorflow.getAllOpList = () => {
+   const opList = OpListProto.decode(tensorflow.GetAllOpList());
+   return opList.op;
+ }
```
Comment (yorkie) [inline lib/index.js]: Don't use arrow function here..

File: src/binding.cc [hunk 2]
```diff
@@ -22,6 +30,8 @@ void InitModule(Handle<Object> target) {
   Nan::Set(target,
     Nan::New<String>("version").ToLocalChecked(), 
     Nan::New<String>(TF_Version()).ToLocalChecked());
+  target->Set(Nan::New("GetAllOpList").ToLocalChecked(),
+               Nan::New<v8::FunctionTemplate>(GetAllOpList)->GetFunction());
```
Comment (yorkie) [inline src/binding.cc]: I think `OpList` should not be here, could we move this method to operation or graph module?

File: src/binding.cc [hunk 3]
```diff
@@ -8,6 +8,14 @@
 #include "src/session.h"
 #include "src/tensor.h"
 
+using namespace v8;
+
+void GetAllOpList(const Nan::FunctionCallbackInfo<Value>& info) {
```
Comment (yorkie) [inline src/binding.cc]: Should wrap it with `NAN_METHOD` :)

**Comment by ry** [comment]
I found this package: https://storage.googleapis.com/tensorflow/libtensorflow/libtensorflow_proto-1.2.1.tar.gz

**Comment by yorkie** [comment]
```xml
<Error>
  <Code>NoSuchKey</Code>
  <Message>The specified key does not exist.</Message>
</Error>
```
Unfortunately got the above error from this link @ry :(

**Comment by ry** [comment]
Sorry. https://storage.googleapis.com/tensorflow/libtensorflow/libtensorflow_proto-1.2.1.zip

**Comment by demohi** [comment]
@yorkie @ry  yes, we can move the protos to https://github.com/tensjs/tensorflow-downloader

**Comment by yorkie** [comment]
@demohi As the #21 has been merged, so that all the protos has been removed, and the OpList should be at `protobuf/tensorflow/core/framework/op_def.proto`. Could you please rebase it with the latest master, thank you :)

**Comment by demohi** [comment]
@yorkie 👍

File: src/binding.cc (hunk 2 above)
Comment (demohi) [inline src/binding.cc]: GetAllOpList is a static method, I think it's not right to put it in operation or graph module. Maybe a utils module ?

File: lib/index.js (hunk 1 above)
Comment (demohi) [inline lib/index.js]: we need a eslint config : )

Comment (yorkie) [inline lib/index.js]: Yea, but manually check before integrating that :)

File: src/binding.cc (hunk 2 above)
Comment (yorkie) [inline src/binding.cc]: What I mean is to move this method on `Graph`, note that that is still a static method. Utils is not suitable for this case IMO.

**Comment by demohi** [comment]
ping @yorkie

**Comment by yorkie** [comment]
CI is passing :)

**Pull request merged**